File: longest-repeating-string.py
class Solution:
    def lengthOfLongestSubstring(self, s: str)-> int:
        chars = [0] * 128
        length = 0
        left = 0

        for right in range( len( s ) ):
            chars[ord(s[right])] += 1

            while( chars[ord(s[right])] > 1 ):
                chars[ord(s[left])] -= 1
                left += 1

            length = max( length, (right - left + 1 ))

        return length

s = "abcabcbb"
#s = "bbbb"
#s = "pwwkew"
sol = Solution()
print(sol.lengthOfLongestSubstring(s))

File: longest-repeating-string.php
<?php

class Solution {
    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s) {
        $hash = [];
        $len = 0;
        $left = 0;

        for( $right = 0; $right < strlen( $s ); $right++) {
            $hash[$s[$right]] = ($hash[$s[$right]]? $hash[$s[$right]]+1: 1);

            while( $hash[$s[$right]] > 1 ) {
                $hash[$s[$left]]--;
                $left++;
            }
            $len = max( $len, ($right - $left + 1 ));
        }
        return $len;
    }
}

$s = "abcabcbb";
//$s = "bbbb";
//$s = "pwwkew";
$sol = new Solution();
echo $sol->lengthOfLongestSubstring($s);